parsl/dataflow/strategy.py

```diff
@@ -275,3 +277,132 @@ def _strategy_simple(self, status_list, tasks, *args, kind=None, **kwargs):
+
+            # reset kill timer if executor has active tasks
+            if active_tasks > 0 and self.executors[executor.label]['idle_since']:
+                self.executors[executor.label]['idle_since'] = None
+
+            # Case 1
+            # No tasks.
+            if active_tasks == 0:
+                # Case 1a
+                # Fewer blocks that min_blocks
+                if active_blocks <= min_blocks:
+                    # Ignore
+                    # logger.debug("Strategy: Case.1a")
+                    pass
+
+                # Case 1b
+                # More blocks than min_blocks. Scale down
+                else:
+                    # We want to make sure that max_idletime is reached
+                    # before killing off resources
+                    if not self.executors[executor.label]['idle_since']:
+                        logger.debug("Executor {} has 0 active tasks; starting kill timer (if idle time exceeds {}s, resources will be removed)".format(
+                            label, self.max_idletime)
+                        )
+                        self.executors[executor.label]['idle_since'] = time.time()
+
+                    idle_since = self.executors[executor.label]['idle_since']
+                    if (time.time() - idle_since) > self.max_idletime:
+                        # We have resources idle for the max duration,
+                        # we have to scale_in now.
+                        logger.debug("Idle time has reached {}s for executor {}; removing resources".format(
+                            self.max_idletime, label)
+                        )
+                        executor.scale_in(active_blocks - min_blocks)
+
+                    else:
+                        pass
+                        # logger.debug("Strategy: Case.1b. Waiting for timer : {0}".format(idle_since))
+
+            # Case 2
+            # More tasks than the available slots.
+            elif (float(active_slots) / active_tasks) < parallelism:
+                # Case 2a
+                # We have the max blocks possible
+                if active_blocks >= max_blocks:
+                    # Ignore since we already have the max nodes
+                    # logger.debug("Strategy: Case.2a")
+                    pass
+
+                # Case 2b
+                else:
+                    # logger.debug("Strategy: Case.2b")
+                    excess = math.ceil((active_tasks * parallelism) - active_slots)
+                    excess_blocks = math.ceil(float(excess) / (tasks_per_node * nodes_per_block))
+                    excess_blocks = min(excess_blocks, max_blocks - active_blocks)
+                    logger.debug("Requesting {} more blocks".format(excess_blocks))
+                    executor.scale_out(excess_blocks)
+
+            elif active_slots == 0 and active_tasks > 0:
+                # Case 4
+                # Check if slots are being lost quickly ?
```

Comment:
what does this comment mean?

Reply:
Stale comment from when the method was copied over. I'll go through and trim old comments.

It meant that if there are >1 tasks and active slots=1, it might be useful to check if the blocks are failing.